File: rb/features.rb
# Quotes SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module QuotesFeatures
  def self.make_feature(name)
    case name
    when "base"
      QuotesBaseFeature.new
    when "ratelimit"
      QuotesRatelimitFeature.new
    when "retry"
      QuotesRetryFeature.new
    when "test"
      QuotesTestFeature.new
    when "timeout"
      QuotesTimeoutFeature.new
    else
      QuotesBaseFeature.new
    end
  end
end

File: py/features.py
# Quotes SDK feature factory

from feature.base_feature import QuotesBaseFeature
from feature.test_feature import QuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: QuotesBaseFeature(),
        "test": lambda: QuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
